Can't build warewulf on a vanilla CentOS 7.9.2009

Following the Quickstart guide, warewulf fails to build.

    $ git clone https://github.com/hpcng/warewulf.git
    Cloning into 'warewulf'...
    remote: Enumerating objects: 5002, done.
    remote: Counting objects: 100% (1219/1219), done.
    remote: Compressing objects: 100% (823/823), done.
    remote: Total 5002 (delta 429), reused 1072 (delta 351), pack-reused 3783
    Receiving objects: 100% (5002/5002), 3.41 MiB | 3.23 MiB/s, done.
    Resolving deltas: 100% (2218/2218), done.
    $ cd warewulf
    $ make all
    golangci/golangci-lint info checking GitHub for tag 'v1.31.0'
    golangci/golangci-lint info found version: 1.31.0 for v1.31.0/linux/amd64
    golangci/golangci-lint info installed .tools/bin/golangci-lint
    get "google.golang.org/protobuf": found meta tag     get.metaImport{Prefix:"google.golang.org/protobuf", VCS:"git", RepoRoot:"https://go.googlesource.com/protobuf"} at //google.golang.org/protobuf?go-get=1
    go: github.com/containers/image/v5@v5.7.0 requires
    	github.com/containers/storage@v1.23.6 requires.
    	github.com/Microsoft/hcsshim@v0.8.9 requires
    	github.com/opencontainers/go-digest@v0.0.0-20180430190053-c9281466c8b2: invalid version: git fetch --unshallow -f origin in     /home/anon/go/pkg/mod/cache/vcs/552be1eeae05d101756f44ffae429e48bc88b11bff177e7eb912ebdd03383c62: exit status 128:
	fatal: git fetch-pack: expected shallow list
    make: *** [vendor] Error 1
    $ lsb_release -a
    LSB Version:	:core-4.1-amd64:core-4.1-noarch:cxx-4.1-amd64:cxx-4.1-noarch:desktop-4.1-amd64:desktop-4.1-noarch:languages-4.1-amd64:languages-4.1-noarch:printing-4.1-amd64:printing-4.1-noarch
    Distributor ID:	CentOS
    Description:	CentOS Linux release 7.9.2009 (Core)
    Release:	7.9.2009
    Codename:	Core
    $

--- a/Makefile
+++ b/Makefile
@@ -12,6 +12,10 @@ GO_TOOLS_VENDOR := $(addprefix vendor/, $(GO_TOOLS))
 GOLANGCI_LINT := $(TOOLS_BIN)/golangci-lint
 GOLANGCI_LINT_VERSION := v1.31.0
 
+# use GOPROXY for older git clients and speed up downloads
+GOPROXY ?= https://proxy.golang.org
+export GOPROXY
+
 # built tags needed for wwbuild binary
 WW_BUILD_GO_BUILD_TAGS := containers_image_openpgp containers_image_ostree
 

